src/groupeddataframe/splitapplycombine.jl

```diff
@@ -717,21 +761,9 @@ function do_f(f, x...)
     end
 end
 
-# Avoid recompilation of larger function for each set of names
-function _combine(@nospecialize(f::NamedTuple{<:Any, <:Tuple{Vararg{Pair}}}),
-                  gd::GroupedDataFrame)
-    idx, valscat = _combine(collect(f), gd)
-    rename!(valscat, collect(Symbol, propertynames(f)))
-    idx, valscat
-end
-
-# Avoid recompilation of larger function for each length
-_combine(@nospecialize(f::Tuple{Vararg{Pair}}), gd::GroupedDataFrame) =
-    _combine(collect(f), gd)
-
 isagg(p::Pair) = check_aggregate(last(p)) isa AbstractAggregate && first(p) isa ColumnIndex
 
-function _combine(f::AbstractVector{<:Pair}, gd::GroupedDataFrame)
+function _combine(f::Vector{Pair}, gd::GroupedDataFrame, nms::Vector{Symbol})
```

Comment:
We generate `f` internally and we know it will be always a `Vector`. Would you think making it `Abstract` would be better.
(a similar case was for `nms` above which you annotated with `AbstractVector` but actually we know it will be a `Vector` because we generate it internally)

Reply:
In general I prefer using the most generic signature which works. Using `Vector` could imply we rely on something more specific, which isn't the case here.

Reply:
Good point - I will change this